python/cudf/cudf/io/parquet.py

```diff
@@ -837,18 +861,19 @@ def write_table(self, df):
             ]
             cw.write_table(grouped_df, this_cw_part_info)
 
-        # Create new cw for unhandled paths encountered in this write_table
-        new_paths, part_info, meta_paths = zip(*new_cw_paths)
-        self._chunked_writers.append(
-            (
-                ParquetWriter(new_paths, **self.common_args),
-                new_paths,
-                meta_paths,
+        if new_cw_paths:
+            # Create new cw for unhandled paths encountered in this write_table
+            new_paths, part_info, meta_paths = zip(*new_cw_paths)
+            self._chunked_writers.append(
+                (
+                    ParquetWriter(new_paths, **self.common_args),
+                    new_paths,
+                    meta_paths,
+                )
```

Comment:
Do you think it would be possible fto wrap these`ParquetWriter` objects in a file-opeing shim? This way, for non-local storaage, we could open the  necessarry paths with fsspec before passing the open "file-like" objects down to `ParquetWriter`. The shim would need to keep track of the open files and close them after the underlying `ParquetWriter` object is closed.

If I understand correctly (which I may not), we *should* be able to write directly to s3 without the file-copy workaround.

Reply:
So this was my initial goto-approach(same as `df.to_parquet`) and after implementing it I ran into issues. Upon investigation, the way `ParquetWriter` is written right now I realized that writing to non-local storage won't fully create and flush the data..even when we manually flush the data after writing a table, only a file is created with 0 bytes and no data is written...for the formation of a correct parquet file with the same file connection open if I try to write the parquet metadata in the footer during the `close` operation the libraries end up overwriting the existing file in `s3` storage and thus losing all the past data and only writing the footer metadata. After repeated tries of different combinations of (write table, flush, write metadata, flush, close) and no one happening successfully I came to a conclusion that the way `s3`(or other cloud file systems) are designed seems to be such that they act as object stores rather than true file-systems where they allow append operations to the contents of existing files. At least for `s3` this is the limitation while `ParquetWriter` is basically built around the concept of `appending`.

Reply:
Sure, will be happy to chat about the findings.